Sources/Sentry/SentryProfiler.mm

```diff
@@ -149,25 +189,46 @@ - (void)start
                     = backtrace_symbols(reinterpret_cast<void *const *>(backtrace.addresses.data()),
                         static_cast<int>(backtrace.addresses.size()));
 #    endif
-                const auto frames = [NSMutableArray<NSDictionary<NSString *, id> *> new];
+
+                const auto stack = [NSMutableArray<NSNumber *> array];
                 for (std::vector<uintptr_t>::size_type i = 0; i < backtrace.addresses.size(); i++) {
-                    const auto frame = [NSMutableDictionary<NSString *, id> dictionary];
-                    frame[@"instruction_addr"] = sentry_formatHexAddress(@(backtrace.addresses[i]));
+                    const auto instructionAddress
+                        = sentry_formatHexAddress(@(backtrace.addresses[i]));
+                    const auto frameIndex = [frames
+                        indexOfObjectPassingTest:^BOOL(NSDictionary<NSString *, id> *_Nonnull obj,
+                            NSUInteger idx, BOOL *_Nonnull stop) {
+                            return [obj[@"instruction_addr"] isEqualToString:instructionAddress];
```

Comment:
`m`: Maybe it makes sense to use a dict with `instructionAddress` as a key for making this `O(1)` instead of `O(n)`? As this is called in a loop, it might be worth the effort.